### DISCUSSION TITLE
Is it possible to generate changelog from PR merge commit?

### DISCUSSION THREAD

**Discussion by seriaati** [orhun/git-cliff#650] [Q&A] [answered]
Say I have a branch `dev` and I commit the following into the branch:
- fix(core): Fix something
- chore(deps): Update dependencies

After that, I open a PR to merge branch `dev` into branch `main`.
Then, I use the "squash and merge" feature to merge the PR to branch `main`.
The `main` branch now has a commit `Merged branch dev to main` or something like that, with the commits in `dev` in the description of the commit.
Now, is it possible to "extract" the commits made in `dev` from that merge commit in `main` and add them into the changelog when I make a changelog in `main`?

**Comment by MeitarR** [comment, marked as answer]
Yes!

git cliff receives a "[commit range](https://darekkay.com/blog/git-commit-ranges/#pull-request-preview)" as a parameter.

so you can do something like

```
git cliff --strip all main..dev
```

and you will get the output you wanted. the `strip all` part is to make sure that no header/footer is added to the output.
